base/pkg/entry.jl

```diff
@@ -197,7 +197,11 @@ function clone(url::AbstractString, pkg::AbstractString)
         end
     catch err
         isdir(pkg) && Base.rm(pkg, recursive=true)
-        rethrow(err)
+        if err.msg == "Unexpected HTTP status code: 404"
```

Comment:
Great, fixed it. Current version checks `isa(err, Base.LibGit2.Error.GitError)` before checking the message field.

Thanks for answering my questions!